enterprise/api/v1/views.py

```diff
@@ -1227,10 +1227,32 @@ def update_branding(self, request, enterprise_uuid):
 
 
 class EnterpriseCustomerCatalogWriteViewSet(EnterpriseWriteOnlyModelViewSet):
+    """
+    API write only views for the ``enterprise-customer-catalog`` API endpoint.
+    """
     queryset = models.EnterpriseCustomerCatalog.objects.all()
     permission_classes = (permissions.IsAdminUser,)
     serializer_class = serializers.EnterpriseCustomerCatalogWriteOnlySerializer
 
+    def create(self, request, *args, **kwargs):
+        """
+        Creates a new EnterpriseCustomerCatalog and returns the created object.
+
+        If an EnterpriseCustomerCatalog already exists for the given enterprise_customer and enterprise_catalog_query,
+        returns the existing object.
+        """
+        enterprise_customer_uuid = request.data.get('enterprise_customer')
+        enterprise_catalog_query_id = request.data.get('enterprise_catalog_query')
+        enterprise_customer_catalog_list = models.EnterpriseCustomerCatalog.objects.filter(
+            enterprise_customer=enterprise_customer_uuid)
+        for catalog in enterprise_customer_catalog_list:
+            catalog_query = catalog.enterprise_catalog_query
+            if catalog_query is not None and catalog_query.id is int(enterprise_catalog_query_id):
+                seralized_customer_catalog = serializers.EnterpriseCustomerCatalogWriteOnlySerializer(
```

Comment:
Since we are expecting this endpoint to be pretty low volume could you add some additional logging in this area to let us know if a new catalog was created or if we returned an existing one?